src/components/Organisms/Rooms/index.tsx

```diff
@@ -2,28 +2,31 @@ import React from 'react'
 import Room from '../../molecules/Room'
 import SkeletonRoom from '../../molecules/SkeletonRoom'
 import { ReactComponent as EmptyStateIcon } from '../../../assets/empty-state.svg'
+import { IRoom } from '../../../types/room'
 
 import styles from './Rooms.module.scss'
 
 type Props = {
-  rooms: {label: string, rounds: number, totalRounds: number, players: number, private: boolean}[]
+  rooms: IRoom[]
   showPrivate: boolean
 }
 
 const Rooms: React.FC<Props> = ({ rooms, showPrivate }) => {
-  return rooms.length > 0
-    ? (
-      <section className={styles.container}>
-        {rooms.filter(room => room?.private === showPrivate).map((room, idx) => {
-          return <Room key={idx} room={room}/>
-        })}
-        <SkeletonRoom/>
-        <SkeletonRoom/>
-        <SkeletonRoom/>
-        <SkeletonRoom/>
-      </section>
-    )
-    : <EmptyStateIcon className={styles['empty-state']}/>
+  const filteredRooms =
+    showPrivate
+      ? rooms.filter(room => room.password)
+      : rooms.filter(room => !Object.prototype.hasOwnProperty.call(room, 'password'))
```

Comment:
Creo que se puede simplificar un poquito:
```suggestion
  const filteredRooms = rooms.filter(room => Boolean(room.password) === showPrivate)
```
o 
```suggestion
  const filteredRooms =
    showPrivate
      ? rooms.filter(room => room.password)
      : rooms.filter(room => !room.password)
```